### PR TITLE
ci: shellcheck checks

### PR DESCRIPTION
**Pull request by storopoli**
Following https://github.com/rust-bitcoin/rust-bitcoin/pull/2762,
adding CI shellcheck cheks here as well.

I also did all fixes that I could find with

```bash
shellcheck **/*.sh
```

If I've missed any please let me know.

### CODE REVIEW

File: contrib/_test.sh
Comment (storopoli) [inline contrib/_test.sh:5]: This is not being used

**Comment by tcharding** [comment]
CI needs re-writing here to use the new `run_task` script.

**Comment by tcharding** [comment]
I don't know what the windows cross problem is, it exists on `master` ATM - I've just disabled the job in this PR and included the error message. Debugging windows builds is not my jam.

**Comment by tcharding** [comment]
Sorry I was not very clear. Mine was just a flippant comment and meant to imply "not worth putting too much effort in here since we need to re-write to use the new scripts".

**Comment by storopoli** [comment]
Oh ok so should I move this to a draft and then rebase later?

**Comment by tcharding** [comment]
You can do that, or if CI is all green I'll just ack it and re-base #699 on top (once I have that one working).

**Comment by storopoli** [comment]
Yeah sorry there's a little delay in shellcheck CI notifications since I don't get them because I've never contributed to this repo.
I'll fix'em.

**Comment by tcharding** [comment]
CI fails are unrelated to this PR. One is the cfg thing same as in rust-bitcoin and one is a windows thing that I've hit elsewhere and have not debugged.

**Comment by storopoli** [comment]
@tcharding let me know when I should rebase this?

**Comment by tcharding** [comment]
You can rebase and force push mate. FTR you can always push changes in front of me, I don't mind rebasing. If something is particularly bothering me I'll explicitly ask it not to merge. Thanks for checking!

**Comment by storopoli** [comment]
rebased, I have not yet made a contribution here, hence I need approval to run CI 😞

**Comment by storopoli** [comment]
Ok CI is still broken unrelated to this PR. I'll rebase sometime in the future when we fix that.

**Comment by apoelstra** [comment]
Weird. CI seems to be failing because of a bug in #709 where we updated the MSRV but didn't update some tests .... but CI passed on #709 itself.

**Comment by apoelstra** [comment]
Oh, no, the failure is the fault of this PR. The `rand-std` and `hashes-std` features do not exist anymore.

**Comment by storopoli** [comment]
> Oh, no, the failure is the fault of this PR. The `rand-std` and `hashes-std` features do not exist anymore.

Good catch! Removed. I need new CI authorization...

**Review by tcharding — APPROVED** [review]
ACK bb3f5003535b9cecee763f36ee9eaef9528260ac

File: secp256k1-sys/depend/secp256k1/ci/ci.sh
Comment (apoelstra) [inline secp256k1-sys/depend/secp256k1/ci/ci.sh:83]: In bb3f5003535b9cecee763f36ee9eaef9528260ac:

If you want to change this file you'll need to PR upstream, or change the vendoring script here to modify it using `sed`. But you can't edit vendored code inline like this.

**Comment by tcharding** [comment]
oooo, bad review by me :)

**Comment by apoelstra** [comment]
Heh, I didn't notice it either. But my local CI scripts did.

FWIW we could change our vendoring code to just delete this file. It's not like we use it.

File: secp256k1-sys/depend/secp256k1/ci/ci.sh
Comment (Kixunil) [reply secp256k1-sys/depend/secp256k1/ci/ci.sh:83]: FTR I believe `$EXTRAFLAGS` is unquoted on purpose.

**Comment by Kixunil** [comment]
We should delete the whole `ci` directory I think.

**Comment by storopoli** [comment]
Ok I removed all changes inside `secp256k1-sys` and added it to the `ignore_paths` according to the [`action-shellcheck` README](https://github.com/ludeeus/action-shellcheck?tab=readme-ov-file#ignore-paths-and-names)

File: .github/workflows/shellcheck.yml
Comment (tcharding) [inline .github/workflows/shellcheck.yml:18]: I'm acking but I don't know this syntax `>-`  - is that just how one makes a newline in yml? I thought it was just `>`

**Review by tcharding — APPROVED** [review]
ACK 52ff1c2e9eae738791ba9df26765fbaf81b6d99a

File: .github/workflows/shellcheck.yml
Comment (storopoli) [reply .github/workflows/shellcheck.yml:18]: Yes also new for me. It was mentioned in the [README of the GH action](https://github.com/ludeeus/action-shellcheck/blob/00b27aa7cb85167568cb48a3838b75f4265f2bca/README.md?plain=1#L47-L57):

> You can use the `ignore_paths` and `ignore_names` input to disable specific 
> directories and files.
> These are passed as environment variables, and should evaluate to a single space-separated string.
> It may be convenient to use [`>-`](https://yaml.org/spec/1.2.2/#65-line-folding) for readability if you have multiple selectors.
> 
> 
> ```text
> sample structure:
> sample/directory/with/files/ignoreme/test.sh
> sample/directory/with/files/ignoremetoo/test.sh
> sample/directory/with/files/test.sh
> sample/directory/with/files/ignorable.sh
> ```

This is the [YAML Line folding](https://yaml.org/spec/1.2.2/#65-line-folding)

**Comment by apoelstra** [comment]
We definitely should not ignore all of `secp256k1-sys`. Our most complicated shellscript (vendor-libsecp.sh) is in there. It's `secp256k1-sys/depend` that we aren't allowed to touch.

File: .github/workflows/shellcheck.yml
Comment (storopoli) [reply .github/workflows/shellcheck.yml]: Yeah let's try to not get smart with YAML. We'll shoot ourselves in the foot. I am not using the `>-` anymore.

File: .github/workflows/shellcheck.yml
Comment (storopoli) [inline .github/workflows/shellcheck.yml:17]: What do you think now, @apoelstra?

Comment (storopoli) [reply .github/workflows/shellcheck.yml:17]: Does it need a comment? Like
```suggestion
        # this is an upstream depency
        ignore_paths: ./secp256k1-sys/depend/**/*.sh
```

Comment (apoelstra) [reply .github/workflows/shellcheck.yml:17]: I think it's fine. If you need to force-push again it'd probably be good to say
```
# This code comes directly from upstream libsecp256k1 and should not be linted here.
```

Comment (storopoli) [reply .github/workflows/shellcheck.yml:19]: Perfect. I (and probably we) do have a bad memory and I think that one extra comment line might save some time in the future.

**Review by apoelstra — DISMISSED** [review]
ACK 888c75d8ecb05bad1ea358be26008538f294c352 successfully ran local tests

**Comment by apoelstra** [comment]
Will let @tcharding take one more look before merging.

**Comment by tcharding** [comment]
ACK b3faab6f91f18259b5eb2ae5fbd1d70d7a9bd4ce

Not that I've caught any mistakes in this PR yet with my reviews :)

File: githooks/pre-commit
Comment (Kixunil) [inline githooks/pre-commit:26]: Why is this needed rather than quoting the command?

Comment (storopoli) [reply githooks/pre-commit:26]: It is the same as https://github.com/rust-bitcoin/rust-bitcoin/blob/5dd20ddfb05754c06aac3f15b9dc6a08574d5167/githooks/pre-commit#L25-L26

Because we would need a nested quoting in L31.

Comment (Kixunil) [reply githooks/pre-commit:26]: What's wrong with nested quoting? `"$(foo "$bar")"` works fine.

Comment (storopoli) [reply githooks/pre-commit:26]: Ok let me try it then...

Comment (storopoli) [reply githooks/pre-commit]: Done in 359a89b1b9a318888036e405cf68f2579720e822

Comment (storopoli) [reply githooks/pre-commit]: Thanks! :)

**Review by Kixunil — DISMISSED** [review]
ACK 359a89b1b9a318888036e405cf68f2579720e822

**Comment by storopoli** [comment]
I cannot decode the error...

**Review by Kixunil — DISMISSED** [review]
ACK bfdf5dc2e00b77b9590c36dd45e6f875012ac3cd

**Comment by Kixunil** [comment]
The CI failure looks unrelated, IIRC we don't pin nightly here.

**Comment by tcharding** [comment]
Maybe just wait for #699 to go in and see if that fixes the CI failure.

File: contrib/extra_tests.sh
Comment (storopoli) [inline contrib/extra_tests.sh:48]: Not being used

File: contrib/sanitizer.sh
Comment (storopoli) [inline contrib/sanitizer.sh:54]: Not being used

**Comment by storopoli** [comment]
rebased after #699 was merged, cc @tcharding

**Review by Kixunil — APPROVED** [review]
ACK ae0a304c2d68efce13c2641d5a3b987bed91d506

**Review by tcharding — APPROVED** [review]
ACK ae0a304c2d68efce13c2641d5a3b987bed91d506

**Review by apoelstra — APPROVED** [review]
ACK ae0a304c2d68efce13c2641d5a3b987bed91d506 successfully ran local tests

**Pull request merged**
